backend/src/main/java/com/github/nickklock/backend/services/SpotService.java

```diff
@@ -0,0 +1,27 @@
+package com.github.nickklock.backend.services;
+
+import com.github.nickklock.backend.models.Spot;
+import com.github.nickklock.backend.models.SpotRequest;
+import com.github.nickklock.backend.repos.SpotRepo;
+import org.springframework.stereotype.Service;
+
+@Service
+public class SpotService {
+
+    private final SpotRepo spotRepo;
+    private final IdService idService;
+
+    public SpotService(SpotRepo spotRepo, IdService idService) {
+        this.spotRepo = spotRepo;
+        this.idService = idService;
+    }
+
+    public Spot add(SpotRequest newSpot) {
+        Spot spotWithId = new Spot(idService.generateId(), newSpot.name(), newSpot.disciplines(), newSpot.waveTypes(),
```

Comment:
Small hint: There is a [cool little pattern for exactly such cases](https://www.baeldung.com/lombok-with-annotations). Unfortunately this is not currently possible with Records via Lombok (as far as I know), but you can build your own helper method with which you can then 

```java
Spot spotWithId = withId(...)
```

[Example](https://github.com/neuefische/java-todo-backend/blob/main/src/main/java/de/neuefische/todo/backend/todo/Todo.java)